overlord/snapstate/check_snap.go

```diff
@@ -433,6 +445,41 @@ func earlyEpochCheck(info *snap.Info, snapst *SnapState) error {
 	return checkEpochs(nil, info, cur, Flags{}, nil)
 }
 
+// check that the listed system users are valid
+var (
+	findUid = osutil.FindUid
+	findGid = osutil.FindGid
+)
+
+func checkSystemUsernames(si *snap.Info) error {
+	if len(si.SystemUsernames) == 0 {
+		return nil
+	}
+	for _, user := range si.SystemUsernames {
+		if !osutil.IsValidUsername(user.Name) {
```

Comment:
this should be done in snap already I think

Reply:
I don't understand this comment. My understanding of this function is it happens extremely early in the snap install checks, so it seemed like a reasonable place to check if the usernames are valid (since validateInfoAndFlags are "lightweight check"s.

Reply:
I  mean if in the code snap/info.go checks for this already we cannot get something invalid here

Reply:
Samuele corrected me and the info check is earlier, so I'll move it.